docs/data-sources/cdn_domains.md

```diff
@@ -0,0 +1,110 @@
+---
+subcategory: Content Delivery Network (CDN)
+---
+
+# huaweicloud_cdn_domains
+
+Use this data source to get a list of CDN domains.
+
+## Example Usage
+
+```hcl
+data "huaweicloud_cdn_domains" "test" {
+  type          = "web"
+  domain_status = "online"
+  service_area  = "mainland_china"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `domain_id` - (Optional, String) Specifies the ID of accelerate domain.
+
+* `name` - (Optional, String) Specifies the name of accelerate domain, using fuzzy matching.
+  Length limit is 1-255 characters.
+
+* `type` - (Optional, String) Specifies the business type of accelerating domain.
+
+* `domain_status` - (Optional, String) Specifies the status of accelerate domain.
+
+* `service_area` - (Optional, String) Specifies the area covered by the acceleration service.
+
+* `enterprise_project_id` - (Optional, String) Specifies the enterprise project ID.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The data source ID.
+
+* `domains` - A list of CDN domains.
+
+The `domains` block supports:
+
+* `id` - The ID of CDN domain.
+
+* `name` - The name of accelerate domain.
+
+* `type` - The business type of accelerating domain.
+
+* `domain_status` - The status of accelerating domain.
+
+* `cname` - The CNAME of the acceleration domain.
+
+* `sources` - An array of one or more objects specifies the domain of the origin server.
+  The [sources](#sources) structure is documented below.
+
+* `domain_origin_host` - The back-to-origin HOST configuration of accelerate domain.
+    The [domain_origin_host](#domain_origin_host) structure is documented below.
+
+* `https_status` - The status of the https.
+
+* `created_at` - The creation time of accelerate domain.
+
+* `updated_at` - The update time of accelerate domain.
+
+* `disabled` - Ban status.
+
+* `locked` - Lock status.
+
+* `auto_refresh_preheat` - Whether to automatically refresh preheating.
+
+* `service_area` - The area covered by the acceleration service.
+
+* `range_based_retrieval_enabled` - Whether to enable range-based retrieval.
+
+* `follow_status` - The status of back-to-source following.
+
+* `origin_status` - Whether to pause origin site return to origin.
+
+* `banned_reason` - The reason why the domain was banned.
+
+* `locked_reason` - The reason why the domain was locked.
+
+* `enterprise_project_id` - The enterprise project ID.
+
+* `tags` - The key/value pairs to associate with the domain.
+
+<a name="sources"></a>
+The `source` block supports:
```

Comment:
Block needs to be consistent with field and structure errors. The best way to write it is: field name=structure name=block name, and the upper and lower anchor points are consistent.